components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,192 @@
+const axios = require("axios");
+const get = require("lodash/get");
+const moment = require("moment");
+const retry = require("async-retry");
+
+module.exports = {
+  type: "app",
+  app: "mailgun",
+  propDefinitions: {
+    domain: {
+      type: "string",
+      label: "domain",
+      description: "Domain used to send and receive email.",
+      async options(context) {
+        const q = context.query;
+        const options = [];
+        const results = await this.getAllDomains();
+        for (const domains of results) {
+          options.push(domains.name);
+        }
+        return options;
+      },
+    },
+    baseRegion: {
+      type: "string",
+      label: "Base region",
+      description:
+        'The region that Mailgun will use to send and receive email, which determines the URL to use when requesting Mailgun API. This must be the same region where "domain" was created.',
+      default: "US",
+      options: ["US", "EU"],
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for events on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // by default, run every 15 minutes.
+      },
+    },
+  },
+  methods: {
+    mailgun(apiKey, domain, host) {
+      return require("mailgun-js")({ apiKey, domain, host });
+    },
+    _apiKey() {
+      return this.$auth.api_key;
+    },
+    _apiHost() {
+      const subdomain = this.baseRegion == "EU" ? "api.eu" : "api";
+      return `${subdomain}.mailgun.net`;
+    },
+    async _makeRequest(opts) {
```

Comment:
To reduce code, I'd recommend destructuring `opts` here:

```javascript
const { domain, method, path, params } = opts
```

so that you can reference the vars directly.

Reply:
Good idea, I made the change. I am still getting to use to argument destructuing\